src/parser.rs

```diff
@@ -1017,10 +1017,31 @@ impl Parser {
             ColumnOption::Unique { is_primary: false }
         } else if self.parse_keyword("REFERENCES") {
             let foreign_table = self.parse_object_name()?;
-            let referred_columns = self.parse_parenthesized_column_list(Mandatory)?;
+            let referred_columns = self.parse_parenthesized_column_list(Optional)?;
+            let mut on_delete = None;
+            let mut on_update = None;
+            while self.parse_keyword("ON") {
+                if self.parse_keyword("DELETE") {
+                    if on_delete == None {
+                        on_delete = Some(self.parse_reference_change_action()?);
+                    } else {
+                        return self
+                            .expected("ON DELETE option not more than once", self.peek_token());
+                    }
+                } else if self.parse_keyword("UPDATE") {
+                    if on_update == None {
+                        on_update = Some(self.parse_reference_change_action()?);
+                    } else {
+                        return self
+                            .expected("ON UPDATE option not more than once", self.peek_token());
+                    }
+                }
```

Comment:
If the above idea doesn't work for some reason, here's an `else` branch missing.